lib/pavilion/plugins/commands/show.py

```diff
@@ -385,65 +412,117 @@ def _functions_cmd(self, _, args):
     def _hosts_cmd(self, pav_cfg, args):
         """List all known host files."""
 
-        hosts = []
-        col_names = ['Name']
-        if args.verbose:
-            col_names.append('Path')
-        for conf_dir in pav_cfg.config_dirs:
-            path = conf_dir / 'hosts'
+        if args.config is None:
+            hosts = []
+            col_names = ['Name']
+            if args.verbose:
+                col_names.append('Path')
+            if args.vars:
+                col_names.append('Variables')
+            for conf_dir in pav_cfg.config_dirs:
+                path = conf_dir / 'hosts'
 
-            if not (path.exists() and path.is_dir()):
-                continue
+                if not (path.exists() and path.is_dir()):
+                    continue
 
-            for file in os.listdir(path.as_posix()):
+                for file in os.listdir(path.as_posix()):
+
+                    file = path / file
+                    if file.suffix == '.yaml' and file.is_file():
+                        host_id = file.stem
+                        host_path = file
+                        with file.open() as config_file:
+                            config_data = yc_yaml.load(config_file)
```

Comment:
Move this logic into a method of the test_config.resolver.Resolver class that finds and loads all of the host or mode files. There's already an equivalent there for test configs.